analysis/test/src/pointers.rs

```diff
@@ -376,6 +376,16 @@ pub unsafe extern "C" fn test_unique_ref() {
     *fresh1 = &mut x as *mut i32;
 }
 #[no_mangle]
+pub unsafe extern "C" fn test_ref_field() {
+    let s = calloc(
+        0i32 as libc::c_ulong,
+        ::std::mem::size_of::<S>() as libc::c_ulong,
+    ) as *mut S;
+    let s_ref = &mut *s;
+    (*s_ref).field4.field4 = (*s_ref).field4.field4;
+    free(s_ref as *mut S as *mut libc::c_void);
```

Comment:
This pattern (`free` on a ptr-casted ref) is surprising to me--I don't think the transpiler emits this sort of code (it would pass the original raw ptr to free), and neither idiomatic nor partially-refactored, transpiled rust seems likely to behave this way either (the latter would presumably not make `s_ref` a mutable reference if it saw it being passed to `free`). This doesn't mean this isn't a reasonable way to test accesses to fields of references for instrumentation, but it did raise some eyebrows. Are we avoiding using a stack `S` here because `AddrOfLocal` support is still WIP?

Reply:
`calloc(0, ...)` here is not sane since we do read/write into the allocation.

Reply:
It should be `1`, right?  How come that doesn't segfault on a null pointer then?